src/main/Config.cpp

```diff
@@ -1427,6 +1476,36 @@ Config::processConfig(std::shared_ptr<cpptoml::table> t)
             throw std::runtime_error(msg);
         }
 
+        if (FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES >
+            PEER_FLOOD_READING_CAPACITY_BYTES)
+        {
+            std::string msg =
+                "Invalid configuration: "
+                "FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES "
+                "can't be greater than PEER_FLOOD_READING_CAPACITY_BYTES";
+            throw std::runtime_error(msg);
+        }
+
+        // This condition is needed to ensure flow control never reaches a
+        // deadlock state. Specifically, it prevents a situation where the
+        // sender is waiting for available capacity to send new messages, but is
+        // unable to do so because the receiver has not yet processed enough
+        // bytes to hit the FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES threshold
+        // and request additional capacity. IOW, by satisfying this condition,
+        // we guarantee that the sender won't be perpetually stalled, waiting
+        // for the receiver's request for more data.
+        if (!(PEER_FLOOD_READING_CAPACITY_BYTES -
+                  FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES >=
+              MAX_CLASSIC_TX_SIZE_BYTES))
```

Comment:
what is max classic tx size? do we read it from somewhere?

Reply:
it's a const. In the future, the actual max tx size value will be derived from this const _and_ Soroban network config.